tsconfig.json

```diff
@@ -24,7 +24,7 @@
 
     "sourceMap": true,
     
-    "allowJs": true,
+    "allowJs": false,
```

Comment:
I've described the problem in the commit message: "TypeScript seems to be confusing JS files with TS files and doing strict checks on things that cannot be typed in JS." At least, VS Code was. Once I turned off this, it stopped doing the strict checks and only showed a subset of the errors.